### PR TITLE
Add `exec` onramp (WIP)

### PR DESCRIPTION
**Pull request by ernadh**

### CODE REVIEW

File: src/onramp/exec.rs [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+    script: &str,
+    args: &Vec<String>,
+    options: &ExecOpts,
+) -> std::result::Result<(Popen, String), ExecError> {
+    match civilize_script(&script.to_string(), &options) {
+        Ok(updated_script) => match create_script_file(&updated_script) {
+            Ok(file) => {
+                let command = match options.shell {
+                    Some(ref value) => value,
+                    None => "sh",
```
Comment (darach) [inline src/onramp/exec.rs:264]: What about windows?

File: src/onramp/exec.rs [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+        file: String,
+        exited: bool,
+        // Exit codes that trigger completion
+        complete_on_code: Vec<i32>,
+        // Marks stream as ended
+        completed: bool
+    }
+}
+
+pub enum ExecType {
```
Comment (darach) [inline src/onramp/exec.rs:63]: Add 'once' semantics to execute once and only once?

File: src/onramp/exec.rs [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+}
+
+impl fmt::Display for ExecType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self)
+    }
+}
+
+pub struct ExecOpts {
+    pub shell: Option<String>,
```
Comment (darach) [inline src/onramp/exec.rs:75]: Set the current working path and ability to set environment variables?

File: src/onramp/exec.rs [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+                                    send_event(
+                                        &pipelines,
+                                        &mut no_pp,
+                                        &mut codec,
+                                        &mut metrics_reporter,
+                                        &mut ingest_ns,
+                                        &origin_uri,
+                                        id,
+                                        // FIXME: This breaks if not UTF8
+                                        std::str::from_utf8(&data).unwrap().into(),
```
Comment (darach) [inline src/onramp/exec.rs:455]: Send an octet sequence rather than convert to utf8 here

**Comment by mfelsche** [comment]
Closing this stale PR.

Feel free to reopen, once the work can progress. 

We miss you @ernadh !

**Pull request closed without merging**